config/config.go

```diff
@@ -582,6 +583,8 @@ type ScrapeConfig struct {
 	MetricsPath string `yaml:"metrics_path,omitempty"`
 	// The URL scheme with which to fetch metrics from targets.
 	Scheme string `yaml:"scheme,omitempty"`
+	// Indicator whether to request compressed response from the target.
+	EnableCompression bool `yaml:"enable_compression"`
```

Comment:
It's usually awkward to add a flag that you have to set to get unchanged behaviour.
So, could this be `disable_compression` ?

Reply:
The value of this flag defaults to `true`, so no, you don't have to set it to get the default behaviour. You need to explicitly set it to `false` to get a different behaviour then the default one.

Reply:
The default is True and I prefer positive flags over negative ones.

Reply:
You've lost me.  When I say "awkward" I mean the ~40 lines like this in this PR:

<img width="504" alt="image" src="https://github.com/prometheus/prometheus/assets/8125524/7f965024-2930-4aff-8090-1730945b676f">

Reply:
This is not user visible, I prefer that for end user we use a enable_ flag.